core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -217,6 +217,11 @@ private String toJson()
         return new JsonRenderer().render(representation);
     }
 
+    public PlanRepresentation getPlanRepresentation()
+    {
+        return representation;
```

Comment:
This shouldn't be be exposed outside this class. It's an internal detail necessary for printing the logical plan. In fact, its initialization should be moved from the constructor to where it's actually used. The distributed plan representation should be handled in a similar manner. Specifically, the distributed plan should not be derived from the PlanRepresentation, but from the SubPlan and other structures.

Reply:
Made that private. 

The PlanRepresentation contains the PlanNode. I would need to call `planRoot.accept()` after creating the PlanRepresentation so that the planRepresentation contains the node representation of all children. That is all done in the constructor.

For building the distributed plan from SubPlan, the PlanFragments' jsonRepresentation is built using PlanRepresentation in `PlanFragmenter::buildFragment`. So, I used the same method when building the DistributedpPlan representation